report_context/controllers/main.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import json
+
+from odoo.http import request, route
+
+from odoo.addons.web.controllers import main as report
+
+
+class ReportController(report.ReportController):
+    @route()
+    def report_routes(self, reportname, docids=None, converter=None, **data):
+        report = request.env["ir.actions.report"]._get_report_from_name(reportname)
+        original_context = json.loads(data.get("context", "{}"))
```

Comment:
> Printing of reports is not working:
> 
> ![imagen](https://user-images.githubusercontent.com/28590170/109012567-712d9100-76b2-11eb-9054-2a6d471f48fd.png)

I've investigated a bit, I think there are two things to consider:

* [ ] We can get data likes this `{"context": None}` so I suggest:
```suggestion
        original_context = json.loads(data.get("context", "{}") or "{}")
```

* [ ] according the dependency tree and current traceback we could get data with context as dict (not str) if report_xlsx is installed; `{"context": {}}`

not sure what's the proper way to achive that maybe fix that in report_xlix to dump that context: https://github.com/OCA/reporting-engine/blob/14.0/report_xlsx/controllers/main.py#L26 or manage some optional dependency to ensure this method is called before if report_xlsx is installed as well (not sure this exist in odoo?).

Reply:
sorry I've written this few weeks ago and didn't realize I've forgot to submit it, I'm not sure if it's still true this was just before @sebastienbeau comment !

Reply:
> sorry I've written this few weeks ago and didn't realize I've forgot to submit it, I'm not sure if it's still true this was just before @sebastienbeau comment !

No problem, thank you for the investigation